Library/Formula/json-fortran.rb

```diff
@@ -0,0 +1,60 @@
+class JsonFortran < Formula
+  desc "A Fortran 2008 JSON API"
+  homepage "https://github.com/jacobwilliams/json-fortran"
+  url "https://github.com/jacobwilliams/json-fortran/archive/4.1.1.tar.gz"
+  sha256 "97f258d28536035ef70e9ead5c7053e654106760a12db2cc652587ed61b76124"
+
+  head "https://github.com/jacobwilliams/json-fortran.git"
+
+  option "with-unicode-support", "Build json-fortran to support unicode text in json objects and files"
+  option "without-test", "Skip running build-time tests (not recommended)"
+  depends_on "cmake" => :run
+  option "without-robodoc", "Do not build and install ROBODoc generated documentation for json-fortran"
+  depends_on "robodoc" => [:recommended, :build]
+  depends_on :fortran
+
+  def install
+    mktemp do
+      args = ["-DUSE_GNU_INSTALL_CONVENTION:BOOL=TRUE"]
```

Comment:
Please see my [note from before](https://github.com/Homebrew/homebrew/pull/40128#discussion_r31146419) attached to the outdated diff.

This is a CMake option to modify where json-fortran files are installed, json-fortran defaults to `FALSE` for this value.

By default (when this CMake cache variable is `FALSE`) the project installs all library archive files and the Fortran `.mod` module file to `<prefix>/jsonfortran-${CMAKE_Fortran_COMPILER_ID}-${VERSION}/lib` and the CMake files used by the `find_package` CMake mechanism get installed into `jsonfortran-${CMAKE_Fortran_COMPILER_ID}-${VERSION}/cmake` This allows multiple versions of the library to be installed for use with different Fortran compilers. (The `.mod` file is like a precompiled header, and this and the object files in the library archives are not compatible between Fortran compilers. Client code MUST link against json-fortran using the same compiler that json-fortran was compiled with.) For example you may have a gfortran compiled version in `/usr/local/jsonfortran-gnu-4.1.1/` and an Intel compiled version in `/usr/local/jsonfortran-intel-4.1.1/`. The CMake `find_package` mechanism can then automatically link against the correct version of the library based on which Fortran compiler is currently being used.

However, IMO it does not make sense to use this install structure with Homebrew for the following reasons:
 - Users should manually install json-fortran if they are planning on supporting multiple Fortran compilers
 - This directory layout does not follow the usual Homebrew layout:
   - Homebrew puts `.mod` files into `include/` AFAICT for all other Fortran packages
   - The CMake files needed by `find_package` are installed into `lib/cmake/`, or at least this what the `scalapack` package seems to do
   - If the default layout is used all files end up installed into `/usr/local/homebrew/Cellar/json-fortran/<version>/jsonfortran-${CMAKE_Fortran_COMPILER_ID}-<version>/…` rather than `/usr/local/homebrew/Cellar/json-fortran/<version>/…` and this prevents the symbolic links from being deployed when `brew link` is run.

Passing `-DUSE_GNU_INSTALL_CONVENTION:BOOL=TRUE` to CMake causes the install directory structure to be more consistent with other Homebrew Fortran packages and GNU conventions at the cost of making it hard or impossible to allow json-fortran to be installed for use with multiple Fortran compilers.